ui/appui-react/src/appui-react/messages/ActivityMessage.tsx

```diff
@@ -26,37 +34,110 @@ export interface ActivityMessageProps {
 
 /** Activity Message React component
  * @public
+ * @deprecated
  */
 export function ActivityMessage(props: ActivityMessageProps) {
   const messageDetails = props.activityMessageInfo.details;
   const [percentCompleteLabel] = React.useState(UiFramework.translate("activityCenter.percentComplete"));
   const [cancelLabel] = React.useState(UiCore.translate("dialog.cancel"));
 
   return (
-    <ToastPresentation
-      category="informational"
-      hasCloseButton={true}
-      onClose={props.dismissActivityMessage}
-      content={
-        <MessageLayout
-          buttons={(messageDetails && messageDetails.supportsCancellation) &&
-            <MessageHyperlink onClick={props.cancelActivityMessage}>{cancelLabel}</MessageHyperlink>
-          }
-          progress={(messageDetails && messageDetails.showProgressBar) &&
+    <Message // eslint-disable-line deprecation/deprecation
+      status={Status.Information}
+      icon={<HollowIcon iconSpec="icon-info-hollow" />}
+    >
+      <MessageLayout
+        buttons={(messageDetails && messageDetails.supportsCancellation) ? (
+          <div>
+            <MessageHyperlink onClick={props.cancelActivityMessage}>
+              {cancelLabel}
+            </MessageHyperlink>
+            <span>&nbsp;</span>
+            <MessageButton onClick={props.dismissActivityMessage}>
+              <Icon iconSpec="icon-close" />
+            </MessageButton>
+          </div>
+        ) : (
+          <MessageButton onClick={props.dismissActivityMessage}>
+            <Icon iconSpec="icon-close" />
+          </MessageButton>
+        )}
+        progress={(messageDetails && messageDetails.showProgressBar) &&
             <MessageProgress
               status={Status.Information}
               progress={props.activityMessageInfo.percentage}
             />
+        }
+      >
+        <div>
+          <MessageLabel message={props.activityMessageInfo.message} className="uifw-statusbar-message-brief" />
+          {(messageDetails && messageDetails.showPercentInMessage) &&
+            <Small>{props.activityMessageInfo.percentage + percentCompleteLabel}</Small>
           }
-        >
-          <div>
-            <MessageLabel message={props.activityMessageInfo.message} className="uifw-statusbar-message-brief" />
-            {(messageDetails && messageDetails.showPercentInMessage) &&
-              <Small>{props.activityMessageInfo.percentage + percentCompleteLabel}</Small>
-            }
-          </div>
-        </MessageLayout>
+        </div>
+      </MessageLayout>
+    </Message>
+  );
+}
+
+interface CustomActivityMessageProps {
+  cancelActivityMessage?: () => void;
+  dismissActivityMessage?: () => void;
+  activityMessageInfo?: ActivityMessageEventArgs;
+  settings?: ToasterSettings;
+}
+
+/**
+ * Hook to render an Activity message.
+ */
+export function useActivityMessage({activityMessageInfo, dismissActivityMessage, cancelActivityMessage, settings}: CustomActivityMessageProps) {
```

Comment:
Hook which calls iTwinUI `toaster` with custom `ActivityMessage` UI. It subsribes to `activityMessages` events to show or close the `ActivityMessage`.